test/ng/compileSpec.js

```diff
@@ -2546,6 +2555,52 @@ describe('$compile', function() {
               template: '<span></span>'
             };
           });
+          directive('prototypeMethodNameAsScopeVarD', function() {
+            return {
+              scope: {
+                'constructor': '<?',
+                'valueOf': '<'
+              },
+              restrict: 'AE',
+              template: '<span></span>'
+            };
+          });
+          directive('optionalBindingsVarA', function() {
```

Comment:
Couldn't you use the `constructor` optional binding of `prototypeMethodAsScopeVar<X>`?